integration-tests/relayinterface/lookups_test.go

```diff
@@ -0,0 +1,431 @@
+	ctx := tests.Context(t)
+	t.Run("AccountLookup resolves valid address with just one address", func(t *testing.T) {
+		expectedAddr := chainwriter.GetRandomPubKey(t)
+		testArgs := chainwriter.TestArgs{
+			Inner: []chainwriter.InnerArgs{
+				{Address: expectedAddr.Bytes()},
+			},
+		}
+		expectedMeta := []*solana.AccountMeta{
+			{
+				PublicKey:  expectedAddr,
+				IsSigner:   true,
+				IsWritable: true,
+			},
+		}
+
+		lookupConfig := chainwriter.AccountLookup{
+			Name:       "TestAccount",
+			Location:   "Inner.Address",
+			IsSigner:   true,
+			IsWritable: true,
+		}
+		result, err := lookupConfig.Resolve(ctx, testArgs, nil, nil)
+		require.NoError(t, err)
+		require.Equal(t, expectedMeta, result)
+	})
+
+	t.Run("AccountLookup resolves valid address with just multiple addresses", func(t *testing.T) {
+		expectedAddr1 := chainwriter.GetRandomPubKey(t)
+		expectedAddr2 := chainwriter.GetRandomPubKey(t)
+
+		testArgs := chainwriter.TestArgs{
+			Inner: []chainwriter.InnerArgs{
+				{Address: expectedAddr1.Bytes()},
+				{Address: expectedAddr2.Bytes()},
+			},
+		}
+		expectedMeta := []*solana.AccountMeta{
+			{
+				PublicKey:  expectedAddr1,
+				IsSigner:   true,
+				IsWritable: true,
+			},
+			{
+				PublicKey:  expectedAddr2,
+				IsSigner:   true,
+				IsWritable: true,
+			},
+		}
+
+		lookupConfig := chainwriter.AccountLookup{
+			Name:       "TestAccount",
+			Location:   "Inner.Address",
+			IsSigner:   true,
+			IsWritable: true,
+		}
+		result, err := lookupConfig.Resolve(ctx, testArgs, nil, nil)
+		require.NoError(t, err)
+		for i, meta := range result {
+			require.Equal(t, expectedMeta[i], meta)
+		}
+	})
+
+	t.Run("AccountLookup fails when address isn't in args", func(t *testing.T) {
+		expectedAddr := chainwriter.GetRandomPubKey(t)
+
+		testArgs := chainwriter.TestArgs{
+			Inner: []chainwriter.InnerArgs{
+				{Address: expectedAddr.Bytes()},
+			},
+		}
+		lookupConfig := chainwriter.AccountLookup{
+			Name:       "InvalidAccount",
+			Location:   "Invalid.Directory",
+			IsSigner:   true,
+			IsWritable: true,
+		}
+		_, err := lookupConfig.Resolve(ctx, testArgs, nil, nil)
+		require.Error(t, err)
+	})
+}
+
+func TestPDALookups(t *testing.T) {
+	programID := solana.SystemProgramID
+
+	t.Run("PDALookup resolves valid PDA with constant address seeds", func(t *testing.T) {
+		seed := chainwriter.GetRandomPubKey(t)
+
+		pda, _, err := solana.FindProgramAddress([][]byte{seed.Bytes()}, programID)
+		require.NoError(t, err)
+
+		expectedMeta := []*solana.AccountMeta{
+			{
+				PublicKey:  pda,
+				IsSigner:   false,
+				IsWritable: true,
+			},
+		}
+
+		pdaLookup := chainwriter.PDALookups{
+			Name:      "TestPDA",
+			PublicKey: chainwriter.AccountConstant{Name: "ProgramID", Address: programID.String()},
+			Seeds: []chainwriter.Lookup{
+				chainwriter.AccountConstant{Name: "seed", Address: seed.String()},
+			},
+			IsSigner:   false,
+			IsWritable: true,
+		}
+
+		ctx := context.Background()
```

Comment:
nit: think we should use a test context for all of these sub-tests `ctx := tests.Context(t)`.

Reply:
fixed